gateway_test.go

```diff
@@ -71,27 +78,21 @@ func (f *MockMetricsFactory) Create(eventName string) Metrics {
 	return metrics
 }
 
-type ForbiddenCheckHttpRequestHandler struct {
-	forbidden string
-}
-
 func mustGetMetricsFactory(t *testing.T, gateway gatewayResource) *MockMetricsFactory {
 	factory, ok := gateway.metricsFactory.(*MockMetricsFactory)
 	if !ok {
-		panic("Failed to get metrics factory")
+		t.Fatal("Failed to get metrics factory")
 	}
 	return factory
 }
 
-func (h ForbiddenCheckHttpRequestHandler) Handle(req *http.Request, metrics Metrics) (*http.Response, error) {
-	if req.Host == h.forbidden {
-		metrics.Fire(metricsResultTargetRequestForbidden)
-		return nil, GatewayTargetForbiddenError
-	}
+type MockHttpRequestHandler struct{}
```

Comment:
Per Go naming conventions for acronyms :(

Note that we already seem to be doing a mix of "Http" and "HTTP", so I don't blame you. However I wanna nudge this code towards Go conventions.


```suggestion
type MockHTTPRequestHandler struct{}
```